react-native-community/cli
# Choose which device to run the app on

Summary:
---------

This PR adds a feature to output a list of available devices when building so that the user can choose which device to run the app on.

### iOS
![Kapture 2020-07-31 at 10 19 21](https://user-images.githubusercontent.com/6207220/89016037-52cc0680-d318-11ea-8689-e51e77fc764a.gif)

### Android

![Kapture 2020-07-31 at 10 22 57](https://user-images.githubusercontent.com/6207220/89016043-552e6080-d318-11ea-845b-204e64126849.gif)

Discussion points:
---------

1. Is this valuable to the CLI?
> The reason I'm submitting this is because I have wanted this for a while and I need to open the IDEs (Xcode & Android Studio) *every single time* I need to choose a different device to run the app on.
1. Is the `--device` flag the best way to make this feature available?
> I'm not sure and I'm definitely open for opinions, I went through the path that doesn't get in the way of the users but can still be valuable, perhaps we could do this under `--choose-device`?

Test Plan:
----------

1. In the CLI folder, run: `yarn link-packages`;
1. In your app project, run: `yarn link "@react-native-community/cli-platform-ios" "@react-native-community/cli-platform-android" "@react-native-community/cli" "@react-native-community/cli-server-api" "@react-native-community/cli-types" "@react-native-community/cli-tools" "@react-native-community/cli-debugger-ui"`;
1. Still on the app project, run `npx react-native start --watchFolders=/path/to/cli/`;
1. And then: `npx react-native run-ios --device` or `npx react-native run-android --device`.

#### Test cases

- Run on an emulator/simulator that has not been started;
- Run on an emulator/simulator that has already been started;
- [Android] Run on an emulator that has not been started and close it as soon as it opens (should show an error);
- Run on a physical device (not properly tested on Android).
